### PR TITLE
[core] add/improved IsPublished/IsSubscribed logic (5.13.x)

### PR DESCRIPTION
**Pull request by rex-schilasky**
### Description
* additional check for CSubscriber::GetPublisherCount() using the connection state of the matching subscriber to ensure that GetPublisherCount is increased only if the matching publisher is able to send data
* test added to test new behavior

### CODE REVIEW

File: ecal/core/src/pubsub/ecal_subgate.cpp [hunk 1]
```diff
@@ -295,8 +295,12 @@ namespace eCAL
 
         iter->second->ApplyLocLayerParameter(process_id, topic_id, tlayer.type(), writer_par);
       }
-      // inform for local publisher connection
-      iter->second->ApplyLocPublication(process_id, topic_id, topic_info);
+      // we only inform the subscriber when the publisher has already recognized at least on local subscriber
```
Comment (KerstinKeller) [inline ecal/core/src/pubsub/ecal_subgate.cpp:298]: at least on -> at least one

File: ecal/core/src/pubsub/ecal_subgate.cpp [hunk 2]
```diff
@@ -345,8 +349,12 @@ namespace eCAL
         const std::string writer_par = tlayer.par_layer().SerializeAsString();
         iter->second->ApplyExtLayerParameter(host_name, tlayer.type(), writer_par);
       }
-      // inform for external publisher connection
-      iter->second->ApplyExtPublication(host_name, process_id, topic_id, topic_info);
+      // we only inform the subscriber when the publisher has already recognized at least on external subscriber
```
Comment (KerstinKeller) [inline ecal/core/src/pubsub/ecal_subgate.cpp:352]: At least on -> at least one

**Review by KerstinKeller — COMMENTED** [review]
Let's remove IsPublished() for 5.13 and add it only for 5.14.

File: ecal/core/src/pubsub/ecal_subgate.cpp (hunk 1 above)
Comment (rex-schilasky) [reply ecal/core/src/pubsub/ecal_subgate.cpp]: fixed

File: ecal/core/src/pubsub/ecal_subgate.cpp (hunk 2 above)
Comment (rex-schilasky) [reply ecal/core/src/pubsub/ecal_subgate.cpp]: fixed

**Pull request merged**